crates/host/src/wasmbus/mod.rs

```diff
@@ -863,33 +970,39 @@ impl ActorInstance {
 
     #[instrument(skip(self, payload))]
     async fn handle_call(&self, payload: impl AsRef<[u8]>) -> anyhow::Result<Bytes> {
-        let Invocation {
-            origin,
-            target,
-            operation,
-            msg,
-            id: invocation_id,
-            ..
-        } = rmp_serde::from_slice(payload.as_ref()).context("failed to decode invocation")?;
+        let invocation: Invocation =
+            rmp_serde::from_slice(payload.as_ref()).context("failed to decode invocation")?;
+
+        debug!(?invocation.origin, ?invocation.target, invocation.operation, "validate actor invocation");
+        invocation.validate_antiforgery(&self.valid_issuers)?;
 
-        debug!(?origin, ?target, operation, "handle actor invocation");
+        // Validate that the actor has the capability to call the target
```

Comment:
You're also calling this in `handle_invocation`. Do we need to check twice?

Reply:
Also great catch